### ISSUE TITLE
Join AND / OR missing space on RAW parameters

### ISSUE THREAD

**Issue by TCB13** [usmanhalalit/pixie#116]
I noticed that a query like:

```
->join(self::$defaultTbl, function($table) use($input, $serviceID, $serviceTbl) {
	$table->on(self::$defaultTbl . ".subscription_id", "=", $table->raw((int)$input->subscription_id));
	$table->on(self::$defaultTbl . ".service_type", "=", $table->raw((int)$serviceID));
	$table->on(self::$defaultTbl . ".service_id", "=", $serviceTbl . ".id");
 })
```

Will output something like:
```
INNER JOIN `Billing\Subscription_Service` ON `Billing\Subscription_Service`.`subscription_id` = 68AND `Billing\Subscription_Service`.`service_type` = 1AND `Billing\Subscription_Service`.`service_id` = `Service\Domain\Service`.`id` 
```

**As we can see there's a missing space here**: `.subscription_id = 68AND Billing\Subs` and  `service_type = 1AND Billing\Subscr`. 

Where's the best way to fix this issue? I manage to trace the creation of the JOIN query to JoinBuilder class. I can add the spaces on to the `AND` directly there. But isn't it better to fix this issue later on where the query is actually built from the statements?

Thank you.

**Comment by TCB13** [comment]
I also find I could fix this by modifying `BaseAdapter.php` (latest version), on:

Line 413:
`$criteria .= " {$statement['joiner']} {$key} {$statement['operator']} $value";`
(added space before `{$statement`)

Line 422:
`$criteria .= ' ' . $statement['joiner'] . ' ' . $key . ' ' . $statement['operator'] . ' ' . $value . ' ';`
(added ` ' ' .` before `$statement['joiner']`

This solution seems more "clean" however I'm not sure if it doesn't have negative impact elsewhere and if there's something nicer.

**Comment by TCB13** [comment]
So I made a experimental fix for this, please check commit d9b6e27

**Comment by TCB13** [comment]
Apparently this was fixed here on PR #114 in a much nicer way!
